src/format/mod.rs

```diff
@@ -756,6 +707,159 @@ fn format_inner<'a>(
+/// `Z` instead of `+00[:]00` is allowed when `allow_zulu` is true.
+#[cfg(any(feature = "alloc", feature = "std", test))]
+fn write_local_minus_utc(
+    result: &mut String,
+    off: FixedOffset,
+    allow_zulu: bool,
+    colon_type: Colons,
+) -> fmt::Result {
+    let mut off = off.local_minus_utc();
+    if !allow_zulu || off != 0 {
+        let mut bytes: [u8; 9] = *b"+00:00:00"; // enough to fit Colons::Double format
+        if off < 0 {
+            off = -off;
+            bytes[0] = b'-';
+        }
+
+        // all formats have the hour offset
+        let hours = off / 3600;
+        bytes[1] = b'0' + (hours / 10) as u8;
+        bytes[2] = b'0' + (hours % 10) as u8;
+
+        let date = match colon_type {
+            Colons::None => {
+                let mins = off / 60 % 60;
+                bytes[3] = b'0' + (mins / 10) as u8;
+                bytes[4] = b'0' + (mins % 10) as u8;
+                &bytes[0..5]
+            }
+            Colons::Single => {
+                // colon is already in the bytes
+                let mins = off / 60 % 60;
+                bytes[4] = b'0' + (mins / 10) as u8;
+                bytes[5] = b'0' + (mins % 10) as u8;
+                &bytes[0..6]
+            }
+            Colons::Double => {
+                // colons are already in the bytes
+                let mins = off / 60 % 60;
+                bytes[4] = b'0' + (mins / 10) as u8;
+                bytes[5] = b'0' + (mins % 10) as u8;
+
+                let secs = off % 60;
+                bytes[7] = b'0' + (secs / 10) as u8;
+                bytes[8] = b'0' + (secs % 10) as u8;
+
+                &bytes[0..9]
+            }
+            Colons::Triple => &bytes[0..3],
+        };
+
+        let date = core::str::from_utf8(date).map_err(|_| fmt::Error)?;
+        result.push_str(date);
+        Ok(())
+    } else {
+        result.push('Z');
+        Ok(())
+    }
+}
+
+#[cfg(any(feature = "alloc", feature = "std", test))]
+pub(crate) fn write_rfc3339(
+    result: &mut String,
+    d: &NaiveDate,
+    t: &NaiveTime,
+    off: FixedOffset,
+) -> fmt::Result {
+    // reuse NaiveDate and NaiveTime `Debug` impls which already print ISO 8601 format.
+    // this is faster in this way.
+    d.write_into(result)?;
+    result.push('T');
+    t.write_into(result)?;
+    write_local_minus_utc(result, off, false, Colons::Single)
+}
+
+#[cfg(any(feature = "alloc", feature = "std", test))]
+pub(crate) fn write_rfc2822(
+    result: &mut String,
+    d: &NaiveDate,
+    t: &NaiveTime,
+    off: FixedOffset,
+) -> fmt::Result {
+    #[cfg(feature = "unstable-locales")]
+    let (short_months, short_weekdays) =
+        (locales::short_months(Locale::POSIX), locales::short_weekdays(Locale::POSIX));
+    #[cfg(not(feature = "unstable-locales"))]
+    let (short_months, short_weekdays) = (
+        &["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"],
```

Comment:
Could these potentially be broken out to a const, which could be shared with the ones at the top of `format_inner`

Reply:
I've split it into a `Locales` struct for now, I'm guessing locale code still needs a lot more love so I don't think I should make much more changes to it